src/callbacks_step/euler_acoustics_coupling_dg2d.jl

```diff
@@ -0,0 +1,91 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+function calc_gradient_c_mean_square!(grad_c_mean_sq, u, mesh,
+                                      equations::AcousticPerturbationEquations2D, dg::DGSEM, cache)
+  @unpack derivative_matrix = dg.basis
+
+  @threaded for element in eachelement(dg, cache)
+    for j in eachnode(dg), i in eachnode(dg)
+      c_mean_sq_x = 0.0 # partial derivative of c_mean square in x direction on the ref. element
+      for ii in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, ii, j, element)
+        c_mean = u_node[6]
+        c_mean_sq_x += derivative_matrix[i, ii] * c_mean^2
+      end
+      grad_c_mean_sq[1, i, j, element] = c_mean_sq_x * cache.elements.inverse_jacobian[element]
+
+      c_mean_sq_y = 0.0 # partial derivative of c_mean square in y direction on the ref. element
+      for jj in eachnode(dg)
+        u_node = get_node_vars(u, equations, dg, i, jj, element)
+        c_mean = u_node[6]
+        c_mean_sq_y += derivative_matrix[j, jj] * c_mean^2
+      end
+      grad_c_mean_sq[2, i, j, element] = c_mean_sq_y * cache.elements.inverse_jacobian[element]
+    end
+  end
+
+  return nothing
+end
+
+
+function calc_acoustic_sources!(acoustic_source_terms, u_euler, u_acoustics, vorticity_mean,
+                                source_region, weights, mesh,
+                                equations::AbstractCompressibleEulerEquations{2}, dg::DGSEM, cache)
+  @unpack derivative_matrix = dg.basis
+  @unpack node_coordinates = cache.elements
+
+  acoustic_source_terms .= zero(eltype(acoustic_source_terms))
+
+  @threaded for element in eachelement(dg, cache)
+    for j in eachnode(dg), i in eachnode(dg)
+      x = get_node_coords(node_coordinates, equations, dg, i, j, element)
+      # Only calculate sources on nodes that lie within the acoustic source region
+      if source_region(x)
+        # Calculate vorticity
```

Comment:
Just a comment. Having a standalone function for `calculate_vorticity` could also be useful elsewhere, e.g., as a pointwise value to export for plotting or as an intermediate step to compute the enstrophy for turbulence simulations

Reply:
That is certainly a very good point. However, as of now I have no idea how to implement this point-wise, given that you need information about the entire element to be able to calculate a gradient. The only viable solution I see at the moment is a function that calculates vorticity for the entire domain.

But if you have further ideas on that, I'd be interested to hear them!

Reply:
Ah, right. I forgot about the gradients. I know in FLUXO we already have the pointwise gradients available because of the `br1/br2` needed for the viscous terms.

Reply:
Good question. I think the only place where I would reasonably search for a such a function manually is in `equations/compressible_euler_2d.jl`. There is precedence, we already stored some DG-specific stuff there for the original implementation of the non-conservative terms for MHD. So this is where I would put the function right now, with a good comment explaining its use and limitations.

Reply:
I added a standalone function for calculating the vorticity in d237297.

Reply:
Great, I'll have a look at it in the coming days!